daemon/daemon.go

```diff
@@ -1486,11 +1486,18 @@ func (h *patchConfig) Handle(params PatchConfigParams) middleware.Responder {
 
 	d := h.daemon
 
+	cfgSpec := params.Configuration
+
+	om, err := option.DaemonOptionLibrary.ValidateConfigurationMap(cfgSpec.Options)
+	if err != nil {
+		msg := fmt.Errorf("Invalid configuration option %s", err)
+		return api.Error(PatchConfigFailureCode, msg)
```

Comment:
Should this be `return api.Error(PatchConfigBadRequestCode, err)` ? I think this replaces line 1512/13 (https://github.com/cilium/cilium/pull/5271/commits/7bd52dacd49cd1edcb51d7df2cc6447ef3b260ad#diff-1a1f3e7ad9b1d7584e2d3e7d0c4c3db9R1512) which is also validating?

Relatedly, line 1510 seems to modify the the map so `om` may have fields that should have been removed? It probably doesn't matter, however, but I'm not sure why we removed that option :/

Reply:
Ah, good finding